aten/src/ATen/native/AdaptiveMaxPooling2d.cpp

```diff
@@ -6,18 +6,20 @@
 namespace at {
 namespace meta {
 TORCH_META_FUNC(adaptive_max_pool2d) (const Tensor& input, IntArrayRef output_size) {
-  for (int64_t i = 0; i < input.ndimension(); i++) {
+  int ndim = input.ndimension();
+  for (int64_t i = 1; i < ndim; i++) {
     TORCH_CHECK(input.size(i) > 0,
-        "adaptive_max_pool2d: expected input to have non-empty spatial dimensions, "
+        "adaptive_max_pool2d(): Expected input to have non-empty spatial dimensions, "
         "but input has sizes ", input.sizes(), " with dimension ", i,
         " being empty");
   }
 
-  TORCH_CHECK((input.ndimension() == 3 || input.ndimension() == 4),
-      "non-empty 3D or 4D (batch mode) tensor expected for input");
+  TORCH_CHECK(ndim == 3 || ndim == 4,
+    "adaptive_max_pool2d(): Expected non-empty 3D or 4D tensor with optional 0-dim batch size, but got: ",
```

Comment:
nit: why not just "Expected 3D or 3D input tensor, but got:" ?

"non-empty" isn't fully accurate now, and I don't think "optional 0-dim batch size" needs to be specified

Same for error messages throughout the PR